packages/asl-projects/client/pages/sections/training.js

```diff
@@ -3,20 +3,30 @@ import { useHistory } from 'react-router-dom';
 import { useSelector, shallowEqual } from 'react-redux';
 import { TrainingSummary } from '@ukhomeoffice/asl-components';
 import { Button } from '@ukhomeoffice/react-components';
+import { compareTrainingRecords } from '../../helpers/trainingRecordsComparison';
+import TrainingSummaryCustom from '../../components/training-summary-custom';
 import Fieldset from '../../components/fieldset';
 import ReviewFields from '../../components/review-fields';
 import ChangedBadge from '../../components/changed-badge';
-
+const { FEATURE_FLAG_TRAINING_RECORD } = require('@asl/service/ui/feature-flag');
 export default function Training(props) {
   const { training, basename, readonly, canUpdateTraining } = useSelector(state => state.application, shallowEqual);
   const project = useSelector(state => state.project);
   const form = useRef(null);
   const history = useHistory();
   const trainingComplete = project['training-complete'];
+  const trainingRecordHighlight = true; //useFeatureFlag(FEATURE_FLAG_TRAINING_RECORD);
   const fields = props.fields.map(f => {
     return f.name === 'training-complete' ? { ...f, type: 'comments-only' } : f;
   });
 
+
+
+  const comparisons = compareTrainingRecords(
+    project.training, // version 0
+    project.trainingHistory // array of { version, trainingRecords }
+  );
```

Comment:
This can be memoised to save work when rerendering

```suggestion
  const comparisons = useMemo(
    () => compareTrainingRecords(
      project.training, // version 0
      project.trainingHistory // array of { version, trainingRecords }
    ),
    [project.training, project.trainingHistory]
  );
```